tests/karate/src/test/java/be/utils/JsonConverter.java

```
@@ -0,0 +1,160 @@
+    Json messageData = Json.of(stringData);
+    String messageDataBase64 = convertJsonToBase64(messageData);
+    messagePart.put("data", messageDataBase64);
+    messagePart.put("sender", senderPk);
+
+    String signature = constructSignature(stringData);
+    messagePart.put("signature", signature);
+    String messageId = hash(messageDataBase64.getBytes(), signature.getBytes());
+
+    if (forcedMessageId && !messageIdForced.isEmpty()) {
+      messageId = this.messageIdForced;
+      messageIdForced = "";
+    }
+    messagePart.put("message_id", messageId);
+    System.out.println("message id is : " + messageId);
+
+    String[] witness = new String[0];
+    messagePart.put("witness_signatures", witness);
+
+    return messagePart;
+  }
+
+  public Map<String, Object> constructParamsField(String channel, String messageDataBase64) {
+    Map<String, Object> paramsPart = new LinkedHashMap<>();
+    paramsPart.put("channel", channel);
+
+    Map<String, Object> messagePart = constructMessageField(messageDataBase64, false);
+    paramsPart.put("message", messagePart);
+
+    return paramsPart;
+  }
+
+  /**
+   * Constructs a valid signature on given data
+   */
+  public String constructSignature(String messageData) {
+    try {
+      // Hex representation of the private key
+      byte[] privateKeyBytes = new byte[privateKeyHex.length() / 2];
+
+      for (int i = 0; i < privateKeyBytes.length; i++) {
+        int index = i * 2;
+
+        int val = Integer.parseInt(privateKeyHex.substring(index, index + 2), 16);
+        privateKeyBytes[i] = (byte) val;
+      }
+      PublicKeySign publicKeySign = new Ed25519Sign(privateKeyBytes);
+      byte[] singie = publicKeySign.sign(messageData.getBytes(StandardCharsets.UTF_8));
+      return Base64.getUrlEncoder().encodeToString(singie);
+
+    } catch (Exception e) {
+      System.out.println("CANNOT SIGN");
+      e.printStackTrace();
+    }
+    // This should never happen
+    return signatureForced;
```

I would prefer to throw an error rather than returning an invalid value.

You can throw the error and remove the try block, or create a new error that is thrown here.